src/phoenix/db/insertion/dataset.py

```diff
@@ -0,0 +1,244 @@
+SpanRowId: TypeAlias = int
+
+
+@dataclass(frozen=True)
+class DatasetCreationEvent(DataModificationEvent):
+    dataset_id: DatasetId
+
+
+async def insert_dataset(
+    session: AsyncSession,
+    name: str,
+    description: Optional[str] = None,
+    metadata: Optional[Mapping[str, Any]] = None,
+    created_at: Optional[datetime] = None,
+) -> Optional[DatasetId]:
+    return await session.scalar(
+        insert(models.Dataset)
+        .values(
+            name=name,
+            description=description,
+            metadata_=metadata or {},
+            created_at=created_at,
+        )
+        .returning(models.Dataset.id)
+    )
+
+
+async def insert_dataset_version(
+    session: AsyncSession,
+    dataset_id: DatasetId,
+    description: Optional[str] = None,
+    metadata: Optional[Mapping[str, Any]] = None,
+    created_at: Optional[datetime] = None,
+) -> Optional[DatasetVersionId]:
+    return await session.scalar(
+        insert(models.DatasetVersion)
+        .values(
+            dataset_id=dataset_id,
+            description=description,
+            metadata_=metadata or {},
+            created_at=created_at,
+        )
+        .returning(models.DatasetVersion.id)
+    )
+
+
+async def insert_dataset_example(
+    session: AsyncSession,
+    dataset_id: DatasetId,
+    span_rowid: Optional[SpanRowId] = None,
+    created_at: Optional[datetime] = None,
+) -> Optional[DatasetExampleId]:
+    return await session.scalar(
+        insert(models.DatasetExample)
+        .values(
+            dataset_id=dataset_id,
+            span_rowid=span_rowid,
+            created_at=created_at,
+        )
+        .returning(models.DatasetExample.id)
+    )
+
+
+class RevisionKind(Enum):
+    CREATE = "CREATE"
+    PATCH = "PATCH"
+    DELETE = "DELETE"
+
+
+async def insert_dataset_example_revision(
+    session: AsyncSession,
+    dataset_version_id: DatasetVersionId,
+    dataset_example_id: DatasetExampleId,
+    input: Mapping[str, Any],
+    output: Mapping[str, Any],
+    metadata: Optional[Mapping[str, Any]] = None,
+    revision_kind: RevisionKind = RevisionKind.CREATE,
+    created_at: Optional[datetime] = None,
+) -> Optional[DatasetExampleRevisionId]:
+    return await session.scalar(
+        insert(models.DatasetExampleRevisions)
+        .values(
+            dataset_version_id=dataset_version_id,
+            dataset_example_id=dataset_example_id,
+            input=input,
+            output=output,
+            metadata_=metadata or {},
+            revision_kind=revision_kind.value,
+            created_at=created_at,
+        )
+        .returning(models.DatasetExampleRevisions.id)
+    )
+
+
+class TableAction(Enum):
+    CREATE = "create"
+    APPEND = "append"
+
+    @classmethod
+    def _missing_(cls, v: Any) -> Optional["TableAction"]:
+        if isinstance(v, str) and v and v.isascii() and not v.islower():
+            return cls(v.lower())
+        return None
+
+
+async def add_table(
```

Comment:
this name is a confusing to me, we're not adding an actual SQL table here right? What "table" abstraction is this referring to?

Reply:
renamed as `add_dataset_examples`